src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -417,6 +397,34 @@ public int Seek(int char0, int char1, int char2)
             }
         }
 
+        private static int FindFirstEqualByte(Vector<byte> chEquals)
+        {
+            // Quasi-tree search
+            var vector64 = Vector.AsVectorInt64(chEquals);
+            for (var i = 0; i < Vector<long>.Count; i++)
+            {
+                var longValue = vector64[i];
+                if (longValue == 0) continue;
+
+                var shift = i << 1;
+                var offset = shift << 2;
+                var vector32 = Vector.AsVectorInt32(chEquals);
```

Comment:
Out of curiosity, have you done any micro benchmarks to confirm that creating `vector32` to determine whether or not to check the first or second half of the long is faster than naively checking up to all 8 bytes for random/evenly distributed `chEquals` inputs?